onnxruntime/core/providers/shared_library/provider_author.h

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Provider authors include this file
```

Comment:
let's see if we can have a better name than provider_author.h

Reply:
I would rather have provider_author.h be something like provider_api.h or provider_interfaces.h
and the current provider_interfaces.h be more like provider_internal_wrappers.h
if I were implementing an EP, I would have thought I should include provider_interfaces.h directly.

Reply:
Yeah, I went back and forth a few times on this. I like provider_api.h, I switched it to that.